client/lib/wp/sync-handler/test/cache-index.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import mockery from 'mockery';
+import ms from 'ms';
+
+/**
+ * Internal dependencies
+ */
+import { RECORDS_LIST_KEY } from '../constants';
+import * as testData from './data';
+import localforageMock from './localforage-mock';
+
+let cacheIndex;
+
+const localData = () => localforageMock.getLocalData();
+const setLocalData = data => localforageMock.setLocalData( data );
+const clearLocal = () => setLocalData( {} );
+
+describe( 'cache-index', () => {
+	before( () => {
+		mockery.enable( {
+			warnOnReplace: false,
+			warnOnUnregistered: false
+		} );
+		mockery.registerSubstitute( 'localforage', 'lib/wp/sync-handler/test/localforage-mock' );
+		( { cacheIndex } = require( '../cache-index' ) );
+	} );
+
+	after( function() {
+		mockery.disable();
+		mockery.deregisterSubstitute( 'localforage' );
+	} );
+
+	beforeEach( clearLocal ); // also do inside nested blocks with >1 test
+
+	describe( '#getAll', () => {
+		beforeEach( clearLocal );
+
+		it( 'should return undefined for empty localforage', ( done ) => {
+			return cacheIndex.getAll()
+				.then( res => {
+					try {
+						expect( res ).to.equal( undefined );
+						done();
+					} catch ( err ) {
```

Comment:
I prefer to omit the parentheses in single-parameter fat-arrow functions. But I think I prefer the parentheses in try/catch blocks. Maybe because try/catch already feels like an unusual syntax for me, so... I dunno, the parentheses make me feel grounded. :smile: